### DISCUSSION TITLE
I would like to know how to resolve the "insufficient subnets remaining in the pool" error.

### DISCUSSION THREAD

**Discussion by mseko** [concourse/concourse#5623] [General]
Hi.

I am using Concourse CI version: v4.2.1.
Currently, I frequently get "insufficient subnets remaining in the pool" error.
I would like to know what causes this error. And I want to resolve this error.

Please help.

Thank you.

**Comment by evanchaoli** [comment]
You may customize container subnet:

1. Create a file, say called `garden_config.ini` on worker:

```
[server]
network-pool = 10.254.0.0/21
```

2. When start `concourse worker`, add option `--garden-config=<path to garden_config.ini>`.

**Reply by skreddy6673** [reply]
@evanchaoli Li can you elaborate the process? I use concourse v5.7.1 and Helm deployments.

**Reply by taylorsilva** [reply]
In the chart you would specify the contents of `garden_config.ini` at the `concourse.worker.garden.config` key:

https://github.com/concourse/concourse-chart/blob/05927d63da28673b14f451d2eb26cdd699737c80/values.yaml#L1408-L1418

EDIT: sorry I didn't read the comment in `values.yaml` correctly. It's a path, so you'll want to have the worker mount a config map that contains the contents of `garden_config.ini`

I think you can use `additionalVolumes` to make a config map: https://github.com/concourse/concourse-chart/blob/05927d63da28673b14f451d2eb26cdd699737c80/values.yaml#L2016-L2035
and then mount that config map to the worker with `additionalMounts`: https://github.com/concourse/concourse-chart/blob/05927d63da28673b14f451d2eb26cdd699737c80/values.yaml#L1967-L1974

**Reply by skreddy6673** [reply]
@taylorsilva ok, if i have 10 workers do i have to manually add `garden_config.ini` file on each worker and then redeploy the concourse instance?

**Reply by taylorsilva** [reply]
See the comment here for how to define a configmap: https://github.com/concourse/concourse-chart/blob/05927d63da28673b14f451d2eb26cdd699737c80/values.yaml#L1651-L1662

**Reply by taylorsilva** [reply]
short answer: nope!

If you're deploying the workers with the chart then they should all be based off the same template. Just deploy using helm again.

**Comment by mseko** [comment]
Thank you.
I will make the settings you told me.
